File: src/main/java/com/app/repository/AccountRepository.java
package com.app.repository;

import com.app.model.Account;
import java.sql.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

public class AccountRepository {
    
    private Connection connection;
    
    public AccountRepository(Connection connection) {
        this.connection = connection;
    }
    
    public Optional<Account> findById(long id) throws SQLException {
        String sql = "SELECT account_id, account_number, ssn, password, tier, verified FROM accounts WHERE account_id = ?";
        try (PreparedStatement stmt = connection.prepareStatement(sql)) {
            stmt.setLong(1, id);
            ResultSet rs = stmt.executeQuery();
            if (rs.next()) {
                Account a = new Account();
                a.setAccountId(rs.getLong("account_id"));
                a.setAccountNumber(rs.getString("account_number"));
                a.setSsn(rs.getString("ssn"));
                a.setPassword(rs.getString("password"));
                a.setTier(rs.getInt("tier"));
                a.setVerified(rs.getBoolean("verified"));
                return Optional.of(a);
            }
        }
        return Optional.empty();
    }
    
    public List<Account> findAll() throws SQLException {
        List<Account> accounts = new ArrayList<>();
        String sql = "SELECT account_id, account_number, ssn, password, tier, verified FROM accounts";
        try (Statement stmt = connection.createStatement()) {
            ResultSet rs = stmt.executeQuery(sql);
            while (rs.next()) {
                Account a = new Account();
                a.setAccountId(rs.getLong("account_id"));
                a.setAccountNumber(rs.getString("account_number"));
                a.setSsn(rs.getString("ssn"));
                a.setPassword(rs.getString("password"));
                a.setTier(rs.getInt("tier"));
                a.setVerified(rs.getBoolean("verified"));
                accounts.add(a);
            }
        }
        return accounts;
    }
}

File: src/main/csharp/Data/AccountRepository.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using App.Models;

namespace App.Data
{
    public class AccountRepository
    {
        private readonly SqlConnection _connection;
        
        public AccountRepository(SqlConnection connection)
        {
            _connection = connection;
        }
        
        public Account FindById(long id)
        {
            using (var cmd = new SqlCommand("SELECT AccountId, AccountNumber, Ssn, Password, Tier, Verified FROM Accounts WHERE AccountId = @id", _connection))
            {
                cmd.Parameters.AddWithValue("@id", id);
                using (var reader = cmd.ExecuteReader())
                {
                    if (reader.Read())
                    {
                        return new Account
                        {
                            AccountId = reader.GetInt64(0),
                            AccountNumber = reader.GetString(1),
                            Ssn = reader.GetString(2),
                            Password = reader.GetString(3),
                            Tier = reader.GetInt32(4),
                            Verified = reader.GetBoolean(5)
                        };
                    }
                }
            }
            return null;
        }
        
        public List<Account> FindAll()
        {
            var accounts = new List<Account>();
            using (var cmd = new SqlCommand("SELECT AccountId, AccountNumber, Ssn, Password, Tier, Verified FROM Accounts", _connection))
            {
                using (var reader = cmd.ExecuteReader())
                {
                    while (reader.Read())
                    {
                        accounts.Add(new Account
                        {
                            AccountId = reader.GetInt64(0),
                            AccountNumber = reader.GetString(1),
                            Ssn = reader.GetString(2),
                            Password = reader.GetString(3),
                            Tier = reader.GetInt32(4),
                            Verified = reader.GetBoolean(5)
                        });
                    }
                }
            }
            return accounts;
        }
    }
}

